File: src/main/java/org/example/keycloak/scim/ScimResourceProviderFactory.java
package org.example.keycloak.scim;

import org.keycloak.Config;
import org.keycloak.models.KeycloakSession;
import org.keycloak.models.KeycloakSessionFactory;
import org.keycloak.services.resource.RealmResourceProvider;
import org.keycloak.services.resource.RealmResourceProviderFactory;

public class ScimResourceProviderFactory implements RealmResourceProviderFactory {
  public static final String ID = "scim";

  @Override
  public String getId() {
    return ID;
  }

  @Override
  public RealmResourceProvider create(KeycloakSession session) {
    return new ScimResourceProvider(session);
  }

  @Override public void init(Config.Scope config) {}
  @Override public void postInit(KeycloakSessionFactory factory) {}
  @Override public void close() {}
}

File: src/main/kotlin/org/example/keycloak/scim/ScimResourceProviderFactory.kt
package org.example.keycloak.scim

import org.keycloak.Config
import org.keycloak.models.KeycloakSession
import org.keycloak.models.KeycloakSessionFactory
import org.keycloak.services.resource.RealmResourceProvider
import org.keycloak.services.resource.RealmResourceProviderFactory

class ScimResourceProviderFactory : RealmResourceProviderFactory {
    companion object {
        const val ID = "scim"
    }

    override fun getId(): String = ID

    override fun create(session: KeycloakSession): RealmResourceProvider =
        ScimResourceProvider(session)

    override fun init(config: Config.Scope) {}
    override fun postInit(factory: KeycloakSessionFactory) {}
    override fun close() {}
}